pkg/core/blockchain.go

```diff
@@ -2370,6 +2370,21 @@ unsubloop:
 	}
 }
 
+func (bc *Blockchain) SubscribeForHeaders(ch chan *block.Header) {
+	bc.subCh <- ch
```

Comment:
Take a look at the receiver side of `bc.subCh` (it's `(bc *Blockchain) notificationDispatcher()`). You need to properly handle headers subscription in the dispatcher so that blockchain is able to share its headers with the world. Headers should be taken from `bcEvent`'s `block` field and distributed to the world right before the block event distribution.

Reply:
See the https://github.com/nspcc-dev/neo-go/issues/2751#issuecomment-1844726731 as far.